docs/source/data_entry.rst

```diff
@@ -99,10 +126,14 @@ The measurements of a material can be located with
 
 Updating Data Model Objects
 ---------------------------
-Runs and specs can be quickly modified in-place and persisted with ``upload``, but templates require more care.
+Runs and specs can be quickly modified in-place and persisted with ``register`` or ``register_all``, but templates require more care.
 In particular, changing the bounds or allowed names/labels of a template could invalidate existing data objects; thus every object on platform must be compared against the desired change.
+If there is no risk that an update could invalidate data (e.g., changing an object name), the template can be updated as per runs and specs.
+
+If such a risk exists (e.g., making bounds more restrictive), ``register`` and ``register_all`` will raise exceptions.
 To attempt such a template update, use :meth:`~citrine.resources.data_concepts.DataConceptsCollection.async_update`.
 If the update is invalid, then the reasons for failure are logged.
+As a convenience, the ``update`` method will first try ``register`` and then will fall-back to ``async_update`` with convenient arguments.
```

Comment:
It's unclear to me if `async_update` needs to be included in the high-level documentation at all, do the default arguments cover most use cases? Maybe adding a separate docs block for cases where things like timeout configuration would be more helpful.

Reply:
That's a fair critique, and something I'd considered.  It fell in there because it felt like I should include `sometimes register will work` and then felt like I should explain what `update` was doing.  Would you prefer I just say `templates can only be updated using update` or leave the `register` v. `update` distinction?

Reply:
I would bias towards recommending `update` for all use cases, not just templates, since the `register` vs `async_update` distinction is essentially an implementation detail.

Reply:
It is, but `register_all` is a serious QoL improvement.  Orders of magnitude speedup in processing objects.  Understanding when you can use it seems important.

Reply:
Ah, ok that is important. What about adding an equivalent `update_all` wrapper?

Reply:
That'd be good, but would require adding a route to the API in order to really get the benefit.  

We could put in a method that filtered the templates from the runs/specs, but then you have a challenge of knowing which to execute first; if the template bounds were increased and the values also were broader, we need to load templates first; if the template bounds shrank and extreme values were dropped from the objects, we need to load the objects first.  Client would require knowledge of everything on platform and some business logic in order to draw those conclusions.

Reply:
Alright, given that context your existing documentation makes sense, thanks Ken.